src/main/java/org/csanchez/jenkins/plugins/kubernetes/volumes/SecretVolume.java

```diff
@@ -29,25 +29,39 @@
 
 import hudson.Extension;
 import hudson.model.Descriptor;
+import io.fabric8.kubernetes.api.model.SecretVolumeSource;
 import io.fabric8.kubernetes.api.model.Volume;
 import io.fabric8.kubernetes.api.model.VolumeBuilder;
 
 public class SecretVolume extends PodVolume {
-
     private String mountPath;
     private String secretName;
+    private String defaultMode;
 
     @DataBoundConstructor
-    public SecretVolume(String mountPath, String secretName) {
+    public SecretVolume(String mountPath, String secretName, String defaultMode) {
         this.mountPath = mountPath;
         this.secretName = secretName;
+        this.defaultMode = defaultMode;
+    }
+
+    public SecretVolume(String mountPath, String secretName) {
+        this(mountPath, secretName, null);
     }
 
     @Override
     public Volume buildVolume(String volumeName) {
+        SecretVolumeSource secretVolumeSource = new SecretVolumeSource();
+        secretVolumeSource.setSecretName(getSecretName());
+
+        if (defaultMode != null) {
```

Comment:
use `StringUtils.isBlank` because the UI will set it to `""` IIRC

Reply:
Perfect thanks